idaes/models_extra/gas_solid_contactors/unit_models/fixed_bed_1D.py

```diff
@@ -0,0 +1,2067 @@
+        if self.config.energy_balance_type != EnergyBalanceType.none:
+            # Solid phase - gas to solid heat transfer
+            @self.Constraint(
+                self.flowsheet().time,
+                self.length_domain,
+                doc="Solid phase - gas to solid heat transfer",
+            )
+            def solid_phase_heat_transfer(b, t, x):
+                return pyunits.convert(
+                    b.solid_phase_heat[t, x],
+                    to_units=units_meta_gas("power") / units_meta_gas("length"),
+                ) * pyunits.convert(
+                    b.solid_properties[t, x]._params.particle_dia,
+                    to_units=units_meta_gas("length"),
+                ) == 6 * b.gas_solid_htc[
+                    t, x
+                ] * (
+                    b.gas_phase.properties[t, x].temperature
+                    - pyunits.convert(
+                        b.solid_properties[t, x].temperature,
+                        to_units=units_meta_gas("temperature"),
+                    )
+                ) * pyunits.convert(
+                    b.solid_phase_area[t, x], to_units=units_meta_gas("area")
+                )
+
+            # Solid phase energy balance
+            # Accumulation equal to heat transfer
+            @self.Constraint(
+                self.flowsheet().config.time,
+                self.length_domain,
+                doc="Solid phase energy holdup constraints",
+            )
+            def solid_energy_holdup_calculation(b, t, x):
+                return b.solid_energy_holdup[t, x] == (
+                    b.solid_phase_area[t, x]
+                    * pyunits.convert(
+                        b.solid_properties[t, x].get_energy_density_terms("Sol"),
+                        to_units=units_meta_solid("energy")
+                        / units_meta_solid("volume"),
+                    )
+                )
+
+            @self.Constraint(
+                self.flowsheet().config.time,
+                self.length_domain,
+                doc="Solid phase energy balances",
+            )
+            def solid_enthalpy_balances(b, t, x):
+                if solid_phase.reaction_package is not None:
+                    return b.solid_energy_accumulation[t, x] / units_meta_solid(
+                        "time"
+                    ) == b.solid_phase_heat[t, x] + -sum(
+                        b.solid_reactions[t, x].reaction_rate[r]
+                        * b.solid_phase_area[t, x]
+                        * pyunits.convert(
+                            b.solid_reactions[t, x].dh_rxn[r],
+                            to_units=units_meta_solid("energy_mole"),
+                        )
+                        for r in solid_phase.reaction_package.rate_reaction_idx
+                    )
+                else:
+                    return b.solid_energy_accumulation[t, x] == b.solid_phase_heat[t, x]
```

Comment:
Similar to comment on line 889. It should be modified to include the heat of adsorption term in the energy balance for cases without reactions (physisorption)

Reply:
See prior comment `solid_material_accumulation`